pkg/tracing/tracing.go

```diff
@@ -67,16 +67,27 @@ type Tracer struct {
 	ot.Tracer
 
 	enabled bool
+
+	// this indirection exists strictly to enable error injection
+	inject func(sm ot.SpanContext, format interface{}, carrier interface{}) error
```

Comment:
I don't love adding stuff like this just to we can tickle code paths for test coverage.